File: src/main/java/io/github/dylmeadows/eontimer/model/timer/Gen5TimerMode.java
package io.github.dylmeadows.eontimer.model.timer;

import io.github.dylmeadows.common.javafx.util.Choice;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum Gen5TimerMode implements Choice {
    STANDARD("Standard"),
    C_GEAR("C-Gear"),
    ENTRALINK("Entralink"),
    ENHANCED_ENTRALINK("Entralink+");

    private final String text;
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/timer/Gen5TimerMode.kt
package io.github.dylmeadows.eontimer.model.timer

import io.github.dylmeadows.common.javafx.util.Choice

enum class Gen5TimerMode(private val text: String) : Choice {
    STANDARD("Standard"),
    C_GEAR("C-Gear"),
    ENTRALINK("Entralink"),
    ENHANCED_ENTRALINK("Entralink+");

    override fun getText(): String {
        return text
    }
}